themes/default/content/blog/esc-kubernetes-cluster-and-app/index.md

```diff
@@ -0,0 +1,317 @@
+            containers:
+            - name: nginx
+              image: "nginx"
+              ports:
+              - containerPort: 80
+  nginx-service:
+    type: kubernetes:core/v1:Service
+    properties:
+      metadata:
+        name: nginx
+      spec:
+        type: LoadBalancer
+        selector:
+          app: nginx
+        ports:
+        - protocol: TCP
+          port: 80
+```
+
+Again, we need to set some configuration before we deploy this stack. In this case, we need a `kubeconfig` to allow
+the `pulumi-kubernetes` provider to connect to our EKS cluster. Let's create another Pulumi ESC environment called
+"kubernetes-cluster" to handle this:
+
+```yaml
+values:
+  stacks:
+    fn::open::pulumi-stacks:
+      stacks:
+        eks-cluster:
+          stack: eks-yaml/demo
+  kubeconfig: {'fn::toJSON': "${stacks.eks-cluster.kubeconfig}"}
+  pulumiConfig:
+    kubernetes:kubeconfig: ${kubeconfig}
+  files:
+    KUBECONFIG: ${kubeconfig}
+```
+
+This ESC environment uses another ESC provider to reference outputs from Pulumi stacks in your organization. In this case,
+we specify that we want to read outputs from the `eks-yaml/demo` stack, and then export the `kubeconfig` output from that
+stack as `pulumiConfig` and `files`. As we saw on the cluster environment, `pulumiConfig` sets configuration for Pulumi
+providers, so this allows us to run our stack and deploy the application to our cluster.
+
+We import this environment with the following stack configuration:
+
+```yaml
+environment:
+  - kubernetes-cluster
+```
+
+We update our stack, and see that our application has been deployed to the cluster.
+
+```
+pulumi up --skip-preview
+Updating (demo)
+
+     Type                              Name               Status
+ +   pulumi:pulumi:Stack               yaml-example-demo  created (0.31s)
+ +   ├─ kubernetes:core/v1:Service     nginx-service      created (10s)
+ +   └─ kubernetes:apps/v1:Deployment  nginx-deployment   created (2s)
+
+Resources:
+    + 3 created
+
+Duration: 13s
+```
+
+Now that the application is deployed, we might want to check some things on the cluster using `kubectl`. To do this, we
+can set the `KUBECONFIG` environment variable to the path of a cluster configuration file. Since we configured this
+in our ESC environment, we can run `kubectl` in the context of this environment like this:
+
+```
+$ pulumi env run k8s -- kubectl cluster-info
+
+Kubernetes control plane is running at https://2F3303BF9E03F8D61099AAB5ED6F31A4.gr7.us-west-2.eks.amazonaws.com
+CoreDNS is running at https://2F3303BF9E03F8D61099AAB5ED6F31A4.gr7.us-west-2.eks.amazonaws.com/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy
+
+$ esc run kubernetes-cluster -- kubectl get deploy
+
+NAME    READY   UP-TO-DATE   AVAILABLE   AGE
+nginx   3/3     3            3           103s
+```
+
+{{% notes type="info" %}}
+ESC environments can be used with the `pulumi` CLI using the [pulumi env](/docs/cli/commands/pulumi_env) command, or with
```

Comment:
@cnunciato Any idea why this link isn't working? https://www.pulumi.com/docs/cli/commands/pulumi_env/ exists, but this is giving me a 404. `/docs/cli/commands` works as expected.

Reply:
For whatever reason, some pages are missing from the preview, specifically each command's page. I observe that, on the preview's sidebar, the `Pulumi CLI | Commands` node has no children. It is assumedly a preview problem.